activestorage/app/models/active_storage/preview.rb

```diff
@@ -75,6 +75,10 @@ def key
     end
   end
 
+  # Downloads the file associated with this preview's variant.
+  # Raises ActiveStorage::Preview::UnprocessedError if the preview has not been processed yet.
+  #
+  # See ActiveStorage::Variant#download for more information.
```

Comment:
I feel like since calling this method without a block could be a footgun, we should copy the [warning from `ActiveStorage::Variant#download`](https://github.com/rails/rails/blob/6442aac195a227455c53c5b5b00c6818268379e5/activestorage/app/models/active_storage/variant.rb#L82-L83) here, rather than hope the user will see it there.

```suggestion
  # Downloads the file associated with this preview's variant. If no block is 
  # given, the entire file is read into memory and returned. That'll use a lot 
  # of RAM for very large files. If a block is given, then the download is 
  # streamed and yielded in chunks. Raises ActiveStorage::Preview::UnprocessedError 
  # if the preview has not been processed yet.
```

What do you think?

Reply:
Good point, it's true that this could be a footgun. I pushed changes to include this. Please have a look.